src/main/java/org/elasticsearch/action/admin/indices/status/TransportIndicesStatusAction.java

```diff
@@ -231,8 +220,8 @@ protected ShardStatus shardOperation(IndexShardStatusRequest request) throws Ela
                     default:
                         stage = GatewayRecoveryStatus.Stage.INIT;
                 }
-                shardStatus.gatewayRecoveryStatus = new GatewayRecoveryStatus(stage, gatewayRecoveryState.getTimer().startTime(), gatewayRecoveryState.getTimer().time(),
-                        gatewayRecoveryState.getIndex().totalByteCount(), gatewayRecoveryState.getIndex().reusedByteCount(), gatewayRecoveryState.getIndex().recoveredByteCount(), gatewayRecoveryState.getTranslog().currentTranslogOperations());
+                shardStatus.gatewayRecoveryStatus = new GatewayRecoveryStatus(stage, recoveryState.getTimer().startTime(), recoveryState.getTimer().time(),
```

Comment:
instead of calling `recoveryState.getIndex()` all time can we assign it to a local var above?